### DISCUSSION TITLE
Persistent Channel enhancement idea

### DISCUSSION THREAD

**Discussion by askids** [dotnet/dotNext#120] [Ideas]
hi,

I had an idea on improving usability of Persistent Channel. Allowing recoverability on a channel is a great feature. But whatif this feature is being used in something like an app running on containers and on multiple instances? In such a case, adding support to leverage something like a distributed cache instead of file based persistence would be more helpful. Something like an Redis OM library that allows to deal with collection or some feature like that with a distribute cache may be more useful with persistent channel.

Thanks!

**Comment by sakno** [comment]
A directory for a persistent channel files can be mounted using NFS, Longhorn, CephFS or any other distributed file system. It's not a goal to reimplement Redis client. However, you can use .NEXT Raft implementation to organize reliable and consistent message exchange between cluster parties.

**Reply by sakno** [reply]
Additionally, it's better to use something like RabbitMQ or MQTT for distributed apps.
